src/analysis/convex/basic.lean

```diff
@@ -254,6 +255,78 @@ begin
   simp [set.mem_image, mem_add, eq_comm]
 end
 
+/-- The translation of a convex set is also convex -/
+lemma convex.translate_preimage_right (hs : convex s) (z : E) : convex ((λ x, z + x) ⁻¹' s) :=
+begin
+  intros x y hx hy a b ha hb hab,
+  rw [mem_preimage],
+  have := calc
+    z + (a • x + b • y) = (a + b) • z + (a • x + b • y)   : by simp only
+                                                                [add_left_inj, hab, one_smul]
+                    ... = a • z + b • z + a • x + b • y   : by simp only [add_smul, add_assoc]
+                    ... = a • z + a • x + b • z + b • y   : by abel
+                    ... = a • (z + x) + b • (z + y)       : by simp only [←smul_add, add_assoc],
+  rw [this],
+  exact hs hx hy ha hb hab,
+end
+
+/-- The translation of a convex set is also convex -/
+lemma convex.translate_preimage_left (hs : convex s) (z : E) : convex ((λ x, x + z) ⁻¹' s) :=
+begin
+  rw [show (λ x, x + z) = (λ x, z + x), by exact funext (λ (x : E), add_comm x z)],
+  exact convex.translate_preimage_right hs z,
```

Comment:
```suggestion
  convert convex.translate_preimage_right hs z,
  simp only [add_comm]
```

I think this is much cleaner, and you can use similar patterns below.

Reply:
Indeed, thanks!